Feature/movie actor background card

baraa0abd opened the pull request:
<img width="499" height="483" alt="image" src="https://github.com/user-attachments/assets/86655d9a-2eeb-4b09-ae64-1551a549318f" />

baraa0abd committed:
feat: add actor and movie background composables

This commit introduces several new composable functions for displaying movie and actor-related backgrounds and overlays:

- `ActorOverlay`: Displays a circular actor image with an optional border.
- `BlurredBackgroundImage`: Shows a blurred background image, supporting Android S+ blur effects.
- `MoviePosterDetailScreen`: A screen composable that combines a blurred background, an overlay, and a central movie poster.
- `PosterCard`: Displays a movie poster with rounded corners and a shadow.

baraa0abd committed:
Merge branch 'feature/similar-movie-screen' into feature/movie-actor-background-card

baraa0abd committed:
Refactor: Delete ActorOverlay composable

The `ActorOverlay` composable function has been removed. This function was previously used to display an actor's image with a circular border and shadow.

baraa0abd committed:
feat: Enhance MoviePosterDetailScreen for actor display

This commit updates the `MoviePosterDetailScreen` composable to support displaying actor images in addition to movie posters.

Key changes:
- Renamed `posterImageUrl` to `ImageUrl` for broader applicability.
- Added an `isActor` boolean parameter to differentiate between movie posters and actor images.
- Implemented conditional rendering:
    - If `isActor` is true, an `ActorOverlay` with a circular shape is displayed.
    - Otherwise, the existing `PosterCard` with rounded corners is displayed.
- Updated the preview to reflect the parameter name change.

baraa0abd committed:
feat: ActorOverlay composable

This commit introduces a new composable function `ActorOverlay`.

This composable displays an actor's image within a circular frame. It includes a shadow effect and a colored border. The image is loaded asynchronously using Coil.

baraa0abd committed:
Merge branch 'develop' into feature/movie-actor-background-card

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/composables/movieActorBackround/ActorOverlay.kt:
```diff
@@ -0,0 +1,43 @@
+            .clip(CircleShape)
+            .border(4.dp, borderColor.copy(alpha = 0.7f), CircleShape)
+    ) {
+        AsyncImage(
```
uses filtereimage instead

baraa0abd committed:
Refactor: Replace AsyncImage with FilteredImage for improved image loading

This commit replaces the `AsyncImage` composable from the Coil library with a custom `FilteredImage` composable in several parts of the application. This change aims to enhance image loading capabilities, potentially by adding filtering or other custom processing to the images.

The affected components are:
- `ActorOverlay`
- `BlurredBackgroundImage`
- `PosterCard`

The `MoviePosterDetailScreen` was also slightly reformatted.
